### ISSUE TITLE
Investigate potential bug introduced by 15716124fa6f061ca21e8f6c5a4907a4b0c52562

### ISSUE THREAD

**Issue by houtanb** [DynareTeam/dynare#1119]
The change introduced in 15716124fa6f061ca21e8f6c5a4907a4b0c52562 was meant to fix the existing bug in `aux_equations` that did not recursively substitute variables in `aux_equations` even though they were substituted in `equations`. We must see if this commit introduced a bug or if there is a bug in the way ramsey code deals with auxiliary equations.

@JohannesPfeifer pointed this out due to the failure of `optimal_policy/Ramsey/ramsey_ex_aux.mod` introduced by this commit

**Comment by JohannesPfeifer** [comment]
The problem can be traced back to the substitution of auxiliary variables of type 4 in the steady state file creation. In the unit test, the problem comes from 

```junk_expectation=EXPECTATION(-1)(c(+1)); ```

In Dynare 4.4.3, we had in the steady state file

```
% Auxiliary equations
ys_(11)=ys_(2);
ys_(23)=ys_(2);
```

The first line assigns to E_{t-1}({c_{t+1}}) the value of ```c```. The second assigns to ```c(+1)``` the value of ```c```.
In the unstable, we have 

```
ys_(11)=ys_(23);
ys_(23)=ys_(2);
```

which, instead of relying on the underlying variables, works recursively. That is, in the unstable version we assign to E_{t-1}({c_{t+1}}) the value of ```c(+1)``` instead of ```c```. But ```c(+1)``` is only assigned the value of ```c``` in the next step.

**Comment by MichelJuillard** [comment]
I will try to solve it as part of #1133

**Comment by JohannesPfeifer** [comment]
Closed by #1133
